### ISSUE TITLE
Add a "sentTime" property as it is done on Android SDK

### ISSUE THREAD

**Issue by Tom3652** [firebase/firebase-ios-sdk#9151]
### Feature proposal

* Firebase Component: Messaging

Hello !  
I am a Flutter developer so i am sorry in advance for the naming conventions / objects that i will describe below, that are probably different from the ones in the native SDKs.

**Problem** : 

I have spoken to a ```FlutterFire``` developer in this [issue](https://github.com/FirebaseExtended/flutterfire/issues/4825#issuecomment-1002956447) about a ```sentTime``` property of the ```RemoteMessage``` object when receiving a notification on an iOS device. 

In Android, when receiving a notification, we receive a ```timestamp``` that is the time when the notification was sent.
In the iOS SDK, it seems that such property does not exist.  
I have contacted Firebase support but they told me that i should see directly with the iOS SDK team as well.

So far as a workaround, we can pass a ```timestamp``` in the payload to do so, but it would much better if it could work like on Android SDK. 
The ```sentTime``` property of the notification is passed in the Android SDK and the FlutterFire developers are able to pass it in the ```firebase_messaging``` plugin for Flutter.

**Request** : 

Please add an equivalent property as this ```sentTime``` (the actual name of this property on Android is probably different but you see the meaning and i know it exists) on the iOS SDK.  

If this property already exists on the iOS SDK, please accept my apologies and i would be glad to know which one it is. 

Thanks in advance !

**Comment by chliangGoogle** [comment]
Thanks for your feedback. We will bring the request to the FM team.

In the meantime, there's a workaround since all the info is passed within the payload. You can access the "sentTime" through the "google.c.a.ts" key in the Message payload.

**Comment by Tom3652** [comment]
Thanks for the quick reply.

The developer said [this](https://github.com/FirebaseExtended/flutterfire/issues/4825#issuecomment-1000437595) regarding this issue. 

It seems that this property is also unavailable in newer iOS versions. 

Can you confirm that this property is still available ?

**Comment by chliangGoogle** [comment]
I just tested and it should be available. Are you using console or sender API to send notifications?

**Comment by Tom3652** [comment]
I am using the sender API and i guess the firebase team for flutter as well :)

**Comment by chliangGoogle** [comment]
@Tom3652 I'm able to reproduce that HTTP v1 Sender API does not provide "google.c.a.ts" send time. I will check with our team and get back to you.

**Comment by Tom3652** [comment]
Hello, is there any news regarding this issue ?  
Thanks !

**Comment by Faiizii** [comment]
I don't believe this issue is still exists in new version of iOS. many values are null. iOS version 14.8.1 and firebase_messaging version is 12.0.1.
![Screenshot 2022-08-07 at 3 49 38 PM](https://user-images.githubusercontent.com/41893474/183287383-d39378be-1abe-4aec-8ee1-c7b491257bc5.png)
.

**Comment by thiagotalma** [comment]
We are already in the year 2023 and until today we do not have a simple solution for an issue created in 2021 :man_facepalming:

**Comment by acheyfaiz** [comment]
> We are already in the year 2023 and until today we do not have a simple solution for an issue created in 2021 🤦‍♂️

agree. i still cant find solution for ios.

**Comment by ChrisWangStairs** [comment]
Hello~ Any update on this issue?
sentTime is still null on newest firebase_messaging 15.2.4

**Comment by Bilal-Basbous** [comment]
> Hello~ Any update on this issue? sentTime is still null on newest firebase_messaging 15.2.4

Yeah, Still getting it null as well :/

**Comment by leojaygoogle** [comment]
tl;dr: the FCM iOS SDK doesn't control the interface of [`UNNotification`](https://developer.apple.com/documentation/usernotifications/unnotification) on iOS. So we cannot add/change its properties. We are reluctant to add data to the [`userInfo` dictionary](https://developer.apple.com/documentation/usernotifications/unnotificationcontent/userinfo) just in case other SDKs might want them.

Longer version:
On Android, when an app receives a notification, its [onMessageReceived](https://firebase.google.com/docs/reference/kotlin/com/google/firebase/messaging/FirebaseMessagingService#onMessageReceived%5C(com.google.firebase.messaging.RemoteMessage%5C)) method is called. The [RemoteMessage](https://firebase.google.com/docs/reference/kotlin/com/google/firebase/messaging/RemoteMessage) object contains a lot of information, including `sentTime`. So it makes sense to send all the information to Android to make sure that the `RemoteMessage` object is properly populated.

While on iOS, the `UNNotification` object doesn't have properties like `sentTime`, `from`, etc. It doesn't make sense for FCM to send such data to the app while the FCM iOS SDK doesn't have a use for it. 

"Why do I see `google.c.sender.id`, `gcm.message_id`, etc in `userInfo` then?" you might ask. Well, these serve a purpose. We use them to implement features like Analytics, [BigQuery data export](https://firebase.google.com/docs/cloud-messaging/understand-delivery?platform=ios#bigquery-data-export), etc. If the notification is sent through the [Notifications Composer](https://console.firebase.google.com/project/_/notification), even more fields are added to `userInfo`. These fields (`google.*`, `gcm.*`) are implementation details. They are not part of the API, and they may change without prior announcement. Please don't depend on implementation details.

Naturally, you next question might be "Okay, what if I do need these fields?". The answer is to explicitly put whatever data you need in your notification's payload, don't rely on FCM's implementation details. For example, in [this method](https://github.com/firebase/flutterfire/blob/14bd67f3e9c6a1dc18ef2daf79053cd906d44d88/packages/firebase_messaging/firebase_messaging/ios/firebase_messaging/Sources/firebase_messaging/FLTFirebaseMessagingPlugin.m#L873-L876), instead of checking `google.c.a.ts`, check `flutterfire.sentTime`(or whatever name you like), and remember to put the `flutterfire.sentTime` key value pair in your notification payload.
